dropwizard-client/src/main/java/io/dropwizard/client/ConfiguredCloseableHttpClient.java

```diff
@@ -0,0 +1,91 @@
+package io.dropwizard.client;
+
+import org.apache.http.HttpHost;
+import org.apache.http.HttpRequest;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.conn.ClientConnectionManager;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.params.HttpParams;
+import org.apache.http.protocol.HttpContext;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+public class ConfiguredCloseableHttpClient implements HttpClient, Closeable {
```

Comment:
Do we really need to create a delegator? Why not make it just a DTO?

`RequestConfig` doesn't change behavior of the client and is needed only as a value object.
We could just create return a DTO in  the `createClient` method and provide 2 build methods.
* First - public for Apache HttpClient users, which returns `CloseableHttpClient`
* Second - package-protected for the Jersey bulder, which returns `ConfiguredCloseableHttpClient`
````java
public CloseableHttpClient build(String name) {
   return buildClientWithRequestConfiguration(name).getClient();
}

ConfiguredCloseableHttpClient buildClientWithRequestConfiguration(String name) {
    final InstrumentedHttpClientConnectionManager manager = createConnectionManager(registry, name);
    return createClient(org.apache.http.impl.client.HttpClientBuilder.create(), manager, name);
}
````